launch/dynamo-run/src/lib.rs

```diff
@@ -318,6 +315,62 @@ pub async fn run(
             }
         }
 
+        // No feature flag because it uses a sub-process, it's very cheap to include
+        Output::Vllm => {
+            if flags.base_gpu_id != 0 {
+                anyhow::bail!("vllm does not support base_gpu_id. Set environment variable CUDA_VISIBLE_DEVICES instead.");
+            }
+            let (py_script, mut child) = match subprocess::start_vllm(
+                local_model.path(),
+                flags.tensor_parallel_size,
+                flags.extra_engine_args.as_deref(),
+            )
+            .await
+            {
+                Ok(x) => x,
+                Err(err) => {
+                    anyhow::bail!("Failed starting vllm sub-process: {err}");
+                }
+            };
+            let cancel_token = cancel_token.clone();
+
+            // Sub-process cleanup
+            extra = Some(Box::pin(async move {
+                cancel_token.cancelled().await;
+
+                // Ask subprocess to stop gracefully
+                if let Some(pid) = child.id() {
+                    unsafe { libc::kill(pid as i32, libc::SIGTERM) };
```

Comment:
Why libc.kill(pid) here over child.kill().await like further below?

Reply:
ref for myself later: https://docs.rs/tokio/latest/tokio/process/struct.Command.html#caveats

Reply:
`child.kill()` is a SIGKILL (`-9`). 

It's odd that there isn't a more gracefully option in either the stdlib or tokio. In fact "TERM", wait N seconds, "KILL" is such a common operation you'd think the whole thing would be in the stdlib.